src/models/converters/data_decoded.rs

```diff
@@ -73,3 +75,80 @@ impl DataDecoded {
         }
     }
 }
+
+impl DataDecoded {
+    pub(super) fn build_address_info_index(
+        &self,
+        info_provider: &mut impl InfoProvider,
+    ) -> Option<HashMap<String, AddressInfo>> {
+        let mut index = HashMap::new();
+        if self.method == MULTI_SEND {
+            if let Some(value_decoded_type) =
+                &self.get_parameter_value_decoded(MULTI_SEND_TRANSACTIONS_PARAM)
+            {
+                match value_decoded_type {
+                    ValueDecodedType::InternalTransaction(transactions) => {
+                        transactions.iter().for_each(|transaction| {
+                            insert_value_into_index(&transaction.to, &mut index, info_provider);
+                            put_parameter_into_index(
```

Comment:
This is theoretically already one level further down, lets see if we see an performance impact.